### ISSUE TITLE
Some fraweworks don't  follow the rules

### ISSUE THREAD

**Issue by joanhey** [TechEmpower/FrameworkBenchmarks#6788]

**Comment by nbrady-techempower** [comment]
Thanks @joanhey 

I’m returning to the office in a couple of weeks and I’ll take a look at this.

**Comment by fafhrd91** [comment]
Seems "just" framework uses const message object for json bench. I am not sure, I don't have much of js knowledge.

https://github.com/TechEmpower/FrameworkBenchmarks/blob/542904f589b63322b8610b82583ce8fdde97f6d0/frameworks/JavaScript/just/tfb.config.js#L49

**Comment by billywhizz** [comment]
i think you are right @fafhrd91. i cannot remember why i did it this way but that certainly wasn't intentional. i will create a PR just to use JSON.stringify on a new object.

**Comment by fafhrd91** [comment]
@billywhizz just still uses const message

@nbrady-techempower 
In last [test run](https://www.techempower.com/benchmarks/#section=test&runid=8b5c71ba-c738-444e-80d1-581f7e650b68&test=query), redkale performs two times more in updates bench. Looks like a bug in framework or cheating.

**Comment by volyrique** [comment]
I would not presume malicious intent, given that the author has been [responsive](https://github.com/TechEmpower/FrameworkBenchmarks/issues/7381#issuecomment-1152538557) to issues like that in the past. Anyway, apparently there is already [a fix](https://github.com/redkale/redkale-plugins/commit/ba719e341d7daf85f42be212523112a5f0607369).

**Comment by joanhey** [comment]
Check the multiple queries for Redkale
https://www.techempower.com/benchmarks/#section=test&runid=8b5c71ba-c738-444e-80d1-581f7e650b68&test=query

It's the first and twice faster than the next fw, and in single query have about 40 faster in front.

**Comment by NinoFloris** [comment]
Redkale seems to be using an entity cache.

The single and multiple queries benchmark rules don't permit this "Use of an in-memory cache of World objects or rows by the application is not permitted." (see rule xii of https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#single-database-query) 

Benchmark code:
https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Java/redkale/src/main/java/org/redkalex/benchmark/BenchmarkService.java#L44-L54

For the single query scenario `source.findAsync` is defined here:
https://github.com/redkale/redkale/blob/961ff7fa225cb05273278e01164d7bcb4e03f947/src/main/java/org/redkale/source/DataSqlSource.java#L2606

And the relevant callee of `source.findListAsync` for multiple queries is defined here (just trace it from findListAsync, it's all the same file):
https://github.com/redkale/redkale/blob/961ff7fa225cb05273278e01164d7bcb4e03f947/src/main/java/org/redkale/source/DataSqlSource.java#L3216

**Comment by volyrique** [comment]
As already discussed in #7899 and #7900, it appears that [requirement 7b](https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#requirements-6) of the cached queries test is frequently violated:
> vii. The implementation may use any caching library or approach desired, with the following caveats. [...] b. The implementation should use a bona-fide cache library or server with features such as read through, a replacement algorithm, and so on. Implementations should not create a plain key-value map of objects. If the platform or framework does not ship with a canonical caching capability, use a caching library or separate-process caching server that is realistic for a production environment. A "reference implementation" would use Google's Guava CacheBuilder for in-VM or Redis for separate-process caching.

In particular, the following implementations seem not to comply:
- `atreugo-prefork`
- `fasthttp-prefork`
- `fiber-prefork`
- `gearbox-prefork`
- `redkale` / `redkale-graalvm`
- `salvo-diesel` / `salvo-pg`
- `viz-diesel` / `viz-pg` / `viz-sqlx`

Note that this list is definitely not exhaustive.

Pinging @abahmed, @chrislearn, @Fenny, @fundon, @redkale, and @savsgio for their input.

**Comment by redkale** [comment]
> As already discussed in #7899 and #7900, it appears that [requirement 7b](https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#requirements-6) of the cached queries test is frequently violated:
> 
> > vii. The implementation may use any caching library or approach desired, with the following caveats. [...] b. The implementation should use a bona-fide cache library or server with features such as read through, a replacement algorithm, and so on. Implementations should not create a plain key-value map of objects. If the platform or framework does not ship with a canonical caching capability, use a caching library or separate-process caching server that is realistic for a production environment. A "reference implementation" would use Google's Guava CacheBuilder for in-VM or Redis for separate-process caching.
> 
> In particular, the following implementations seem not to comply:
> 
> * `atreugo-prefork`
> * `fasthttp-prefork`
> * `fiber-prefork`
> * `gearbox-prefork`
> * `redkale` / `redkale-graalvm`
> * `salvo-diesel` / `salvo-pg`
> * `viz-diesel` / `viz-pg` / `viz-sqlx`
> 
> Note that this list is definitely not exhaustive.
> 
> Pinging @abahmed, @chrislearn, @Fenny, @fundon, @redkale, and @savsgio for their input.

[Cache](https://github.com/TechEmpower/FrameworkBenchmarks/pull/8037/files#diff-13155fc1ea3437749139d45cef2b234e8d572e9feac69df388b7524c70c43a48)  had changed

**Comment by redkale** [comment]
@NinoFloris 
Redkale does not use entity caching default, and EntityCache requires the entity class tag @ Cacheable to take effect, such as [CachedWorld](https://github.com/TechEmpower/FrameworkBenchmarks/pull/8037/files#diff -ffa4e74a7b69b72a295dbe6a28097a0e8a97b0a7a457863a7b1beb73b0444974)   @Cacheable(direct = true)

**Comment by cesco69** [comment]
just seem cache request/response https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/JavaScript/just/techempower.js#L76C5-L76C15

**Comment by billywhizz** [comment]
> just seem cache request/response https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/JavaScript/just/techempower.js#L76C5-L76C15

no. it doesn't. this is the cache used for the cached-query benchmark, which is not even included in the composite scores.

**Comment by NateBrady23** [comment]
Closing as part of bulk cleanup.
